sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -522,6 +523,15 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
                         List<Object> inList = (List<Object>) negatedPredicate.get(operator);
                         placeholder = placeholdersForInList(inList, sqlParameters);
+                    } else if (operator.equals(MOD)) {
+                        // The predicate map value must be a two element list containing integers
+                        // here.  This was validated during normalization.
+                        List<Integer> modulus = (List<Integer>) negatedPredicate.get(operator);
+                        // Casting the remainder as integer ensures proper processing of the modulo
+                        // operation by SQLite.  If left un-cast, unexpected results occur.
+                        placeholder = String.format("? %s CAST(? AS INTEGER)", operatorMap.get(EQ));
```

Comment:
That is correct.  This is an issue with how we communicate with SQLite rather than SQLite itself.  On the Android/Java side it looks like the remainder bind argument is being passed in as a String to SQLite.  So the SQL statement that SQLite tries to process looks something like `SELECT _id FROM table WHERE field % 2 = "1"`, which will never return records.  Adding the CAST ensures that the remainder is handled as an integer.  This does not happen on the iOS/Obj C side.